Mathlib/Data/Polynomial/Splits.lean

```diff
@@ -430,6 +424,16 @@ theorem splits_iff_exists_multiset {f : K[X]} :
     splits_of_exists_multiset i hs⟩
 #align polynomial.splits_iff_exists_multiset Polynomial.splits_iff_exists_multiset
 
+theorem splits_of_comp (j : L →+* F) {f : K[X]} (h : Splits (j.comp i) f)
+    (roots_mem_range : ∀ a ∈ (f.map (j.comp i)).roots, a ∈ j.range) : Splits i f := by
```

Comment:
Would it make sense to first prove this theorem without `i` (i.e., set `i = id`) and then deduce this theorem by specializing to `f.map i`?

Reply:
Almost the same proof works for 
```
theorem splits_id_of_splits {f : K[X]} (h : Splits i f)
    (roots_mem_range : ∀ a ∈ (f.map i).roots, a ∈ i.range) : Splits (RingHom.id K) f := sorry
```
but I need an additional `RingHom.comp_id` in the `conv_lhs => rw` block. So I'd rather deduce it from the `comp` version.